core/shared/src/main/scala/matryoshka/patterns/ListF.scala

```diff
@@ -52,6 +52,14 @@ object ListF {
     case NilF()      => Seq.empty
   } (s => s.headOption.fold[ListF[A, Seq[A]]](NilF())(ConsF(_, s.tail)))
 
+  def takeUpTo[N, T, A](implicit N: Recursive.Aux[N, Option], T: Recursive.Aux[T, ListF[A, ?]]): Coalgebra[ListF[A, ?], (N, T)] =
+    pair => pair._1.project.fold[ListF[A, (N, T)]](NilF())(p => pair._2.project.map((p, _)))
+
+  def find[A](cond: A => Boolean): Algebra[ListF[A, ?], Option[A]] = {
```

Comment:
There are common patterns we see in the connectors for example, with `interpretM` and `freeCata`. 

A good approach could be to ask the question: "What questions do people answer with matryoshka?" For example: "I'd like to find all instances of a certain pattern and accumulate those (mapped to another type perhaps) into a list. How do I do that?" And then we wrap that all up in a function that calls `para` that takes as input the pattern to match on.